### PR TITLE
rename Cartesian to Semigroupal

### LINKED ISSUE #1931: Renaming Cartesian to Semigroupal
`Cartesian` is our name for an associative `def product[A, B](fa: F[A], fb: F[B]): F[(A, B)]`. I find this name very confusing given that cartesian *categories* are not the same thing as monoidal and semigroupal *functors*, and there's no reason to associate the action of a semigroupal functor with a cartesian product or zipping seeing as they have much more meaning semantically and aren't actually provided by most instances of Cartesian as it stands. I can see some doubt in the previous PR, https://github.com/typelevel/cats/pull/795, as to whether or not this would be called a "semigroupal" functor. As far as I can tell it is; google "semigroupal functor" for examples, one of which is https://www.researchgate.net/publication/2100971_Abelian_Categories_of_Modules_over_a_Lax_Monoidal_Functor which suggests a definition by analogy to monoidal functors.

With all of that in mind, I'm calling for us to rename `Cartesian` to `Semigroupal` before 1.0.0-RC1 is out. I don't know if there's a convincing case to be made for `Monoidal` to be added.

Attribution: @ekmett at scalaworld told me in passing that the name was incorrect when I mentioned it.

### PR DESCRIPTION
**Pull request by kailuowang**
fixes #1931 with a deprecated type alias and scalafix
There are two commits in this PR the de6345e is the mechanical rename one - not much to look here. 
the [f42419b](https://github.com/kailuowang/cats/commit/f42419b90afb9c659ae3ea0c2ff658d1bf5887eb) adds a scalafix rule that should be easy to review separately

### CODE REVIEW

File: laws/src/main/scala/cats/laws/discipline/InvariantMonoidalTests.scala [hunk 1]
```diff
@@ -24,12 +24,12 @@ trait InvariantMonoidalTests[F[_]] extends InvariantTests[F] with CartesianTests
...
-      val parents = Seq(invariant[A, B, C], cartesian[A, B, C])
+      val parents = Seq(invariant[A, B, C], semigroupal[A, B, C])
       val bases = Seq.empty
       val props = Seq(
-        "invariant cartesian left identity" -> forAll((fa: F[A], b: B) => laws.invariantMonoidalLeftIdentity(fa, b)),
-        "invariant cartesian right identity" -> forAll((fa: F[A], b: B) => laws.invariantMonoidalRightIdentity(fa, b)),
-        "invariant cartesian associativity" -> forAll((fa: F[A], fb: F[B], fc: F[C]) => laws.invariantMonoidalAssociativity(fa, fb, fc))
+        "invariant semigroupal left identity" -> forAll((fa: F[A], b: B) => laws.invariantMonoidalLeftIdentity(fa, b)),
+        "invariant semigroupal right identity" -> forAll((fa: F[A], b: B) => laws.invariantMonoidalRightIdentity(fa, b)),
+        "invariant semigroupal associativity" -> forAll((fa: F[A], fb: F[B], fc: F[C]) => laws.invariantMonoidalAssociativity(fa, fb, fc))
```
Comment (edmundnoble) [inline laws/src/main/scala/cats/laws/discipline/InvariantMonoidalTests.scala]: "Invariant monoidal left identity"?

File: scalafix/README.md [hunk 2]
```diff
@@ -50,3 +50,13 @@ sbt scalafix github:typelevel/cats/v1.0.0
 - [ ] foldLeftM is removed from Free, use foldM on Foldable instead, see #1117 for detail.
 
 - [ ] iteratorFoldM was removed from Foldable due to #1716
+
+
+## To test scala fix
+
+```bash
+sbt ;coreJVM/publishLocal;coreFree/publishLocal
```
Comment (edmundnoble) [inline scalafix/README.md]: Is this redundant with the other scalafix PR?

File: scalafix/output/src/main/scala/fix/v1_0_0/RenameCartesian.scala [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package fix
+package to1_0_0
+
+import cats.Semigroupal
+
+object RenameCartesianTests {
+  import cats.syntax.semigroupal._
+
+  def prod[F[_]: Semigroupal, A, B](fa: F[A], fb: F[B]): F[(A, B)] = Semigroupal[F].product(fa, fb)
```
Comment (edmundnoble) [inline scalafix/output/src/main/scala/fix/v1_0_0/RenameCartesian.scala]: :heart:

File: scalafix/rules/src/main/scala/fix/Cats_v1_0_0.scala [hunk 4]
```diff
@@ -225,4 +225,19 @@ case class RenameEitherTLiftT(index: SemanticdbIndex)
       "_root_.cats.data.EitherTFunctions.liftT." -> "liftF"
     )
 
-}
\ No newline at end of file
+}
+
+// ref: https://github.com/typelevel/cats/pull/1961
```
Comment (edmundnoble) [inline scalafix/rules/src/main/scala/fix/Cats_v1_0_0.scala]: :+1:

File: scalafix/README.md (hunk 2 above)
Comment (kailuowang) [reply scalafix/README.md]: Are you referring the PR that add this to travis? This is to add it to readme so people know how to run the test locally.

File: core/src/main/scala/cats/package.scala [hunk 5]
```diff
@@ -90,4 +90,7 @@ package object cats {
   val Semigroup = cats.kernel.Semigroup
   val Monoid = cats.kernel.Monoid
   val Group = cats.kernel.Group
+
+  @deprecated("renamed to Semigroupal", "1.0.0-RC1")
+  type Cartesian[F[_]] = Semigroupal[F]
```
Comment (LukaJCB) [inline core/src/main/scala/cats/package.scala]: I think you've got some extra white space here

**Comment by kailuowang** [comment]
I'm going to go ahead and merge this. It got 2 sign-offs (1 of which dismissed by minor changes )

**Pull request merged**